src/ssl.c

```diff
@@ -18800,7 +18773,10 @@ WOLF_STACK_OF(WOLFSSL_X509)* wolfSSL_set_peer_cert_chain(WOLFSSL* ssl)
     if ((ssl == NULL) || (ssl->session.chain.count == 0))
         return NULL;
 
-    sk = wolfSSL_sk_X509_new();
+    if (ssl->peerCertChain == NULL)
+        sk = wolfSSL_sk_X509_new();
+    else /* Try to re-use old chain if available */
```

Comment:
If reusing, should the stack be emptied before proceeding? What if there are items in the stack?

Reply:
You're right. This doesn't make sense. I'll just free the existing struct if it exists.